src/sentry/static/sentry/app/views/organizationEventsV2/table.tsx

```diff
@@ -74,9 +74,13 @@ export default class Table extends React.Component<Props> {
 
             let sortKey: string | null = field;
             if (SPECIAL_FIELDS.hasOwnProperty(field)) {
-              sortKey = SPECIAL_FIELDS[field].sortField ? field : null;
+              sortKey = SPECIAL_FIELDS[field as keyof typeof SPECIAL_FIELDS].sortField
+                ? field
+                : null;
             } else if (FIELD_FORMATTERS.hasOwnProperty(field)) {
-              sortKey = FIELD_FORMATTERS[field].sortField ? field : null;
+              sortKey = FIELD_FORMATTERS[field as keyof typeof FIELD_FORMATTERS].sortField
```

Comment:
Why is this syntax needed to look up a value in a object when it is being used as a mapping/lookup table?

Reply:
Good question. `field` has the type `string`. 

Since `FIELD_FORMATTERS.hasOwnProperty(field)` is `true`, we can restrict `field` to be one of the keys of `typeof FIELD_FORMATTERS`.

Otherwise just `FIELD_FORMATTERS[field]` would be an `any` type from TS's perspective.

Reply:
Thanks for explaining that. 🙇